### PR TITLE
feat(starfish): adds synchronized axis markers to starfish charts

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
adds synchronized axis markers to starfish charts

### CODE REVIEW

**Review by wmak — APPROVED** [review]
LGTM 🚢

**Review by shruthilayaj — APPROVED** [review]
can we also add this to the web service view?
the error rate chart isn't using the starfish chart
component rn, but i'm fixing that in one of my PRs
that I have up

**Comment by edwardgou-sentry** [comment]
> can we also add this to the web service view? the error rate chart isn't using the starfish chart component rn, but i'm fixing that in one of my PRs that I have up

👍 will do this in another pr!

**Review by gggritso — COMMENTED** [review]
Ooh, nice!

File: static/app/views/starfish/components/chart.tsx [hunk 1]
```diff
@@ -186,6 +201,28 @@ function Chart({
...
+  const formatter: TooltipFormatterCallback<TopLevelFormatterParams> = (
+    params,
+    asyncTicket
+  ) => {
+    // Kinda jank. Get hovered dom elements and check if any of them are the chart
+    const hoveredEchartElement = Array.from(document.querySelectorAll(':hover')).find(
+      element => {
+        return element.classList.contains('echarts-for-react');
+      }
+    );
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/views/starfish/components/chart.tsx:213]: So this is one of the annoying parts of echarts. When linking charts together, echarts also calls/renders the tooltip formatter for all charts as if you're hovering over them. We really only want to render the tooltip on the chart that the user is hovering over.

What's happening in this formatter code is we're grabbing the echarts dom element and checking to see if it matches the element being hovered. If yes, then we just render the tooltip as usual. If no, then the formatter just returns an empty string, ie doesn't render any tooltip.

**Pull request merged**